### PR TITLE
Add text field component for num of vacancies

### PR DESCRIPTION
**Pull request by alexbokii**
**Change log:**
1) Add component TextField
2) Create a new view for an "About the role" form
3) Use new TextField component to display Immediate start and Future start fields.

! to get to this component, go to http://localhost:8080/exercise/new/information-about-role.
As this journey should start on a View Vanancy page and View Vanancy page doesn't exist yet, we display form in a separate view which will be connected to View Vanancy page in the future.

### CODE REVIEW

File: src/components/Form/TextField.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    </label>
+    <span 
+      v-if="hint"
+      id="event-name-hint" 
+      class="govuk-hint"
+    >
+      {{ hint }}
+    </span>
+    <input 
+      id="key"
```
Comment (ollietreend) [inline src/components/Form/TextField.vue]: Should this be `:id="id"` so it dynamically matches the `:for="id"` on the label?

File: src/components/Form/TextField.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  <div class="govuk-form-group">
+    <label
+      :for="id"
+      class="govuk-heading-m"
+    >
+      {{ label }}
+    </label>
+    <span 
+      v-if="hint"
+      id="event-name-hint"
```
Comment (ollietreend) [inline src/components/Form/TextField.vue]: If this `id` attribute is hardcoded, it'll cause duplicate IDs on the same page.

Do we need an ID on this element? Or can we make it dynamic by using the `id` prop and appending 'hint'? e.g. `:id="id + '_hint'"`

File: tests/unit/components/Form/TextField.spec.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+         expect(wrapper.find('.govuk-hint').text()).toBe('Hint for the label!');
+      }); 
+
+      it('does not show hint if hint is not passed', () => {
+        expect(wrapper.find('.govuk-hint').exists()).toBe(false);
+      }); 
+    });
+
+    describe('id', () => {
+      it('sets the label for attribute', () => {
```
Comment (ollietreend) [inline tests/unit/components/Form/TextField.spec.js]: Could there also be a test that the `<input>` has the correct `id` attribute?

File: tests/unit/components/Form/TextField.spec.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        wrapper.setProps({ id: 'my_unique_key' });
+        expect(wrapper.find('label').attributes().for).toBe('my_unique_key');
+      }); 
+    });
+
+    describe('type', () => {
+      it('sets CSS class is narrow is true', () => {
```
Comment (ollietreend) [inline tests/unit/components/Form/TextField.spec.js]: Typo:

> sets CSS class **if** narrow is true

i.e. change 'is' to 'if'

File: tests/unit/components/Form/TextField.spec.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        wrapper.setProps({ type: 'narrow' });
+        expect(wrapper.find('input').attributes().class).toContain('govuk-input--width-5');
+      });
+
+      it("doesn't set CSS class is narrow is not passed", () => {
```
Comment (ollietreend) [inline tests/unit/components/Form/TextField.spec.js]: Same typo here – change 'is' to 'if'

File: src/components/Form/TextField.vue (hunk 2 above)
Comment (alexbokii) [reply src/components/Form/TextField.vue]: I've just realised, we don't need id at all for this <span>, so I'll remove it for now

**Pull request merged**